### PR TITLE
Validate the ref that was checked out was a tag

### PR DESCRIPTION
**Pull request by keith**
Previously if you had a repo with a branch and a tag with the same name,
using `git clone --branch` would prefer the branch, which may or may not
be the same source as the tag. Now we checkout the tag unambiguously.
The biggest downside here is we no longer do a shallow clone in this
case.

### CODE REVIEW

**Comment by keith** [comment]
I've updated the test git repo so there is a branch named v1.0 that is not at the same place as the tag, this validates the new behavior with the tag checkout test.

File: lib/cocoapods-downloader/git.rb [hunk 1]
```diff
@@ -77,6 +77,10 @@ def clone(force_head = false, shallow_clone = true)
         ui_sub_action('Git download') do
           begin
             git! clone_arguments(force_head, shallow_clone)
+            if !force_head && !validate_tag
+              target_git 'checkout', '--quiet', "tags/#{options[:tag]}"
```
Comment (segiddins) [inline lib/cocoapods-downloader/git.rb]: This might require also fetching the tag? eg refs/tags/...

Comment (keith) [reply lib/cocoapods-downloader/git.rb]: yea definitely, need to figure out how to successfully test that

**Comment by dnkoutso** [comment]
I think we should land this, its a good check

**Pull request closed without merging**